vignettes/intro.Rmd

```diff
@@ -0,0 +1,162 @@
+---
+title: "A Brief Introduction to RNeXML"
+author:
+  - Carl Boettiger
+  - Scott Chamberlain
+  - Rutger Vos
+  - Hilmar Lapp
+
+output: html_vignette
+---
+
+<!--
+%\VignetteEngine{knitr::rmarkdown}
+%\VignetteIndexEntry{A Brief Introduction to RNeXMLL}
+-->
+
+
+```{r echo=FALSE}
+library(RNeXML)
+```
+
+Read in a `nexml` file into the `ape::phylo` format:
+
+```{r}
+f <- system.file("examples", "comp_analysis.xml", package="RNeXML")
+nexml <- nexml_read(f)
+tr <- get_trees(nexml) # or: as(nexml, "phylo")
+plot(tr)
+```
+
+Write an `ape::phylo` tree into the `nexml` format:
+
+```{r}
+data(bird.orders)
+nexml_write(bird.orders, "test.xml")
+```
+
+A key feature of NeXML is the ability to formally validate the construction of the data file against the standard (the lack of such a feature in nexus files had lead to inconsistencies across different software platforms, and some files that cannot be read at all).  While it is difficult to make an invalid NeXML file from `RNeXML`, it never hurts to validate just to be sure:
+
+```{r}
+nexml_validate("test.xml")
+```
+
+
+
+Extract metadata from the NeXML file: 
+
+```{r } 
+birds <- nexml_read("test.xml")
+get_taxa(birds)
+get_metadata(birds) 
+```
+
+--------------------------------------------
+
+
+Add basic additional metadata:  
+
+```{r}
+  nexml_write(bird.orders, file="meta_example.xml",
+              title = "My test title",
+              description = "A description of my test",
+              creator = "Carl Boettiger <cboettig@gmail.com>",
+              publisher = "unpublished data",
+              pubdate = "2012-04-01")
+
+```
+By default, `RNeXML` adds certain metadata, including the NCBI taxon id numbers for all named taxa.  This acts a check on the spelling and definitions of the taxa as well as providing a link to additional metadata about each taxonomic unit described in the dataset.  
+
+
+### Advanced annotation
+
+
+We can also add arbitrary metadata to a NeXML tree by define `meta` objects:
+
+```{r}
+modified <- meta(property = "prism:modificationDate",
+                 content = "2013-10-04")
+```
+
+Advanced use requires specifying the namespace used.  Metadata follows the RDFa conventions.  Here we indicate the modification date using the prism vocabulary. This namespace is included by default, as it is used for some of the basic metadata shown in the previous example.  We can see from this list:
+
+```{r}
+RNeXML:::nexml_namespaces
+```
+
+This next block defines a resource (link), described by the `rel` attribute as a homepage, a term in the `foaf` vocabulalry.  Becuase `foaf` is not a default namespace, we will have to provide its URL in the full definition below. 
```

Comment:
```suggestion
This next block defines a resource (link), described by the `rel` attribute as a homepage, a term in the `foaf` vocabulalry.  Because `foaf` is not a default namespace, we will have to provide its URL in the full definition below. 
```